backends/test/harness/tester.py

```diff
@@ -0,0 +1,453 @@
+import random
+from collections import Counter, OrderedDict
+from typing import Any, Dict, List, Optional, Tuple, Type
+
+import torch
+
+from executorch.backends.test.harness.stages import (
+    Export,
+    Partition,
+    Quantize,
+    RunPasses,
+    Serialize,
+    Stage,
+    StageType,
+    ToEdge,
+    ToEdgeTransformAndLower,
+    ToExecutorch,
+)
+from executorch.exir.dim_order_utils import get_memory_format
+
+from torch.export import ExportedProgram
+from torch.testing import FileCheck
+
+
+class Tester:
+    """
+    Base class for a backend tester. This class is not intended to be used directly. Instead,
+    backends are expected to subclass it and provide implementations for backend-dependent
+    stages.
+    """
+
+    def __init__(
+        self,
+        module: torch.nn.Module,
+        example_inputs: Tuple[torch.Tensor],
+        stage_classes: Dict[StageType, Type],
+        dynamic_shapes: Optional[Tuple[Any]] = None,
+    ):
+        module.eval()
+
+        self.stage_classes = stage_classes
+        self.original_module = module
+        self.example_inputs = example_inputs
+        self.dynamic_shapes = dynamic_shapes
+        self.stages: Dict[StageType, Stage] = OrderedDict.fromkeys(list(StageType))
+        self.pipeline = {
+            StageType.QUANTIZE: [StageType.EXPORT],
+            StageType.EXPORT: [
+                StageType.RUN_PASSES,
+                StageType.TO_EDGE,
+                StageType.TO_EDGE_TRANSFORM_AND_LOWER,
+            ],
+            StageType.TO_EDGE_TRANSFORM_AND_LOWER: [
+                StageType.RUN_PASSES,
+                StageType.TO_EXECUTORCH,
+            ],
+            StageType.TO_EDGE: [
+                StageType.PARTITION,
+                StageType.RUN_PASSES,
+            ],
+            StageType.RUN_PASSES: [
+                StageType.PARTITION,
+                StageType.TO_EDGE_TRANSFORM_AND_LOWER,
+            ],
+            # TODO Make this Stage optional
```

Comment:
can you just do this by making TO_EDGE either go to TO_EXECUTORCH or PARTITION. Also I know its a bit of a headache now, but I wash oping to change the name of partition to TO_BACKEND. If its not too much of a hassle to do. otherwise it's a very minor nit